core/dbt/parser/schemas.py

```diff
@@ -878,12 +898,68 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
                     return
 
         elif patch.yaml_key == "analyses":
-            unique_id = self.manifest.analysis_lookup.get_unique_id(patch.name, None)
+            unique_id = self.manifest.analysis_lookup.get_unique_id(patch.name, None, None)
         else:
             raise DbtInternalError(
                 f"Unexpected yaml_key {patch.yaml_key} for patch in "
                 f"file {source_file.path.original_file_path}"
             )
+
+        # patch versioned nodes
+        versions = block.target.versions
+        if versions:
```

Comment:
this is the meat and potatoes of the change. Currently it's in `NonSourceParser`, which is subclassed by `TestablePatchParser` for parsing models, seeds, and snapshots. Given the increasing number of model-specific patching that's emerged as part of the 'Models as APIs' work (access, constraints, now versions) - I'm thinking this could be split out into a new subclass for parsing models patches independently of seeds and snapshots. Perhaps simply `ModelPatchParser`. What do you think @gshank, cc @peterallenwebb?